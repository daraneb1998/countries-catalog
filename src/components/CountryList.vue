<script setup lang="ts">
import { useFetchCountries } from "@/services/useFetchCountries";
import CountryItem from "./CountryItem.vue";
import { computed } from "vue";

const people = [
  {
    name: "Calvin Hawkins",
    email: "calvin.hawkins@example.com",
    image:
      "https://images.unsplash.com/photo-1491528323818-fdd1faba62cc?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80",
  },
  {
    name: "Kristen Ramos",
    email: "kristen.ramos@example.com",
    image:
      "https://images.unsplash.com/photo-1550525811-e5869dd03032?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80",
  },
  {
    name: "Ted Fox",
    email: "ted.fox@example.com",
    image:
      "https://images.unsplash.com/photo-1500648767791-00dcc994a43e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80",
  },
  {
    name: "Calvin Hawkins",
    email: "calvin.hawkins@example.com",
    image:
      "https://images.unsplash.com/photo-1491528323818-fdd1faba62cc?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80",
  },
  {
    name: "Kristen Ramos",
    email: "kristen.ramos@example.com",
    image:
      "https://images.unsplash.com/photo-1550525811-e5869dd03032?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80",
  },
  {
    name: "Ted Fox",
    email: "ted.fox@example.com",
    image:
      "https://images.unsplash.com/photo-1500648767791-00dcc994a43e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80",
  },
];
const { error, countries, isFetching } = useFetchCountries();
console.log("countries", countries);
</script>

<template>
  <div
    class="absolute top-[18vh] overflow-y-scroll w-screen h-[79vh] flex flex-row justify-center py-4 my-4"
  >
    <ul
      class="grid grid-cols-2 md:grid-cols-3 xl:grid-cols-4 2xl:grid-cols-5 gap-4 mx-4"
    >
      <li v-for="country in countries" :key="country.flags.png">
        <CountryItem :country="country" :key="country.flags.png" />
      </li>
    </ul>
  </div>
</template>
