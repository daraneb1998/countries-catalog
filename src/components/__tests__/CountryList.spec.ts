import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import CountryList from "../CountryList.vue";

describe("CountryList", () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it("renders properly", () => {
    const wrapper = mount(CountryList);
    expect(wrapper.text()).toMatchSnapshot();
  });
});
