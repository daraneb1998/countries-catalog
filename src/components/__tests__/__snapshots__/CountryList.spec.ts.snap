// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CountryList > renders properly 1`] = `""`;

exports[`SortOption > renders properly 1`] = `"SORT NAMEASCENDINGDESCENDING"`;
